services/movie_session.py

```diff
@@ -0,0 +1,46 @@
+from django.db.models import QuerySet
+
+from django.utils.datetime_safe import datetime
+
+from db.models import MovieSession, CinemaHall, Movie
+
+
+def create_movie_session(movie_show_time: datetime,
+                         movie_id: int, cinema_hall_id: int) -> MovieSession:
+    cinema_hall = CinemaHall.objects.get(id=cinema_hall_id)
+    movie = Movie.objects.get(id=movie_id)
+    return MovieSession.objects.create(
+        show_time=movie_show_time,
+        movie=movie,
+        cinema_hall=cinema_hall
+    )
+
+
+def get_movies_sessions(session_date: str = None) -> None | QuerySet:
+    if session_date:
+        return MovieSession.objects.filter(show_time__date=session_date)
+    return MovieSession.objects.all()
+
+
+def get_movie_session_by_id(movie_session_id: int) -> MovieSession:
+    return MovieSession.objects.get(id=movie_session_id)
+
+
+def update_movie_session(session_id: int, show_time: datetime = None,
+                         movie_id: int = None,
+                         cinema_hall_id: int = None) -> MovieSession:
+    movie_session = get_movie_session_by_id(session_id)
+
+    if show_time:
+        movie_session.show_time = show_time
+    if movie_id:
+        movie_session.movie_id = movie_id
+    if cinema_hall_id:
+        movie_session.cinema_hall_id = cinema_hall_id
+
+    movie_session.save()
+    return movie_session
+
+
+def delete_movie_session_by_id(session_id: int) -> MovieSession:
+    return MovieSession.objects.get(id=session_id).delete()
```

Comment:
The `delete_movie_session_by_id` function returns the result of the `delete()` method, which is a tuple (number of objects deleted, a dictionary with details). If you intend to return the deleted `MovieSession` object, you should retrieve it before calling `delete()`. Otherwise, update the return type hint to reflect the actual return value.